pkg/collectors/devices/pmc.go

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package devices
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/callbacks"
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/fetcher"
+)
+
+type PMCInfo struct {
+	TimeSource              string `json:"timeSource"`
+	ClockAccuracy           string `json:"clockAccuracy"`
+	OffsetScaledLogVariance string `json:"offsetScaledLogVariance"`
+	ClockClass              int    `json:"clockClass"`
+	CurrentUtcOffset        int    `json:"currentUtcOffset"`
+	Leap61                  int    `json:"leap61"`
+	Leap59                  int    `json:"leap59"`
+	CurrentUtcOffsetValid   int    `json:"currentUtcOffsetValid"`
+	PtpTimescale            int    `json:"ptpTimescale"`
+	TimeTraceable           int    `json:"timeTraceable"`
+	FrequencyTraceable      int    `json:"frequencyTraceable"`
+}
+
+type GMSettings struct {
+	PMC PMCInfo `fetcherKey:"pmc" json:"pmc"`
+}
+
+// GetAnalyserFormat returns the json expected by the analysers
+func (gmSetting *GMSettings) GetAnalyserFormat() ([]*callbacks.AnalyserFormatType, error) {
+	formatted := callbacks.AnalyserFormatType{
+		ID: "pmc",
+		Data: []any{
+			gmSetting.PMC,
+		},
```

Comment:
It will also make the json output incorrect it should be values in an array or and an object (once my PR in vse-sync-pp gets merged. The current Will be an array with an object in it.

Reply:
Could you please explain this?